### PR TITLE
Python: Add Server-side Request Forgery sinks

### PR DESCRIPTION
**Pull request by haby0**
Add some sinks in the CodeQL-Python SSRF model. Modeled Python libraries are: `Aiohttp`, `Httpx`, `Libtaxii`, `Urllib`, `Urllib2`, `Urllib3`.

There may be inappropriate modeling, please point out the project maintainer so that I can improve it.

### CODE REVIEW

File: python/ql/lib/semmle/python/frameworks/Aiohttp.qll [hunk 1]
```diff
@@ -639,3 +639,54 @@ module AiohttpWebModel {
...
+        result = this.getArgByName("url")
+        or
+        not methodName = "request" and
+        result = this.getArg(0)
+        or
+        methodName = "request" and
+        result = this.getArg(1)
+      }
+
+      override DataFlow::Node getAUrlPart() { result = this.getUrlArg() }
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiohttp.qll:681]: I don't see any reason to have this separate predicate, can we just do

```suggestion
      override DataFlow::Node getAUrlPart() {
        result = this.getArgByName("url")
        or
        not methodName = "request" and
        result = this.getArg(0)
        or
        methodName = "request" and
        result = this.getArg(1)
      }
```

File: python/ql/lib/semmle/python/frameworks/Aiohttp.qll [hunk 2]
```diff
@@ -639,3 +639,54 @@ module AiohttpWebModel {
...
+
+      override string getFramework() { result = "aiohttp.ClientSession" }
+
+      override predicate disablesCertificateValidation(
+        DataFlow::Node disablingNode, DataFlow::Node argumentOrigin
+      ) {
+        none()
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiohttp.qll:688]:
```suggestion
        // TODO: Look into disabling certificate validation
        none()
```

File: python/ql/lib/semmle/python/frameworks/Aiohttp.qll [hunk 3]
```diff
@@ -639,3 +639,54 @@ module AiohttpWebModel {
     override DataFlow::Node getValueArg() { result = value }
   }
 }
+
+/**
+ * Provides models for the web server part (`aiohttp.client`) of the `aiohttp` PyPI package.
+ * See https://docs.aiohttp.org/en/stable/client.html
+ */
+module AiohttpClientModel {
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiohttp.qll:647]:
```suggestion
private module AiohttpClientModel {
```

**Review by RasmusWL — CHANGES_REQUESTED** [review]
Thanks for this contribution :+1:

Please do the same rewrites I have suggested for `python/ql/lib/semmle/python/frameworks/Aiohttp.qll` to the other files as well.

Also please add tests to highlight the cases we can now handle, like in https://github.com/github/codeql/blob/main/python/ql/test/library-tests/frameworks/requests/test.py

File: python/ql/lib/semmle/python/frameworks/Aiohttp.qll (hunk 2 above)
Comment (haby0) [reply python/ql/lib/semmle/python/frameworks/Aiohttp.qll:688]: Ok

File: python/ql/lib/semmle/python/frameworks/Aiohttp.qll (hunk 1 above)
Comment (haby0) [reply python/ql/lib/semmle/python/frameworks/Aiohttp.qll]: > Also please add tests to highlight the cases we can now handle, like in https://github.com/github/codeql/blob/main/python/ql/test/library-tests/frameworks/requests/test.py

Please review whether the test file is correct, thank you!

Comment (RasmusWL) [reply python/ql/lib/semmle/python/frameworks/Aiohttp.qll]: They looked great, but there was a failure for `httpx`. Next time you should run the tests in all directories before submitting :wink:

**Comment by RasmusWL** [comment]
Thanks for the updates @haby0. Since you did not target `experimental/`, I've had to ensure that everything is 100% good looking before merging this PR, so I have incorporated my changes on this PR directly.

**Comment by haby0** [comment]
> Thanks for the updates @haby0. Since you did not target `experimental/`, I've had to ensure that everything is 100% good looking before merging this PR, so I have incorporated my changes on this PR directly.

@RasmusWL 👍 
I tested the test files in the `codeql\python\ql\test\experimental\library-tests` path, and they were all executed successfully

**Comment by RasmusWL** [comment]
Performance looks good :+1:

**Review by RasmusWL — APPROVED** [review]

**Comment by haby0** [comment]
> Performance looks good 👍

Thanks.

**Pull request merged**